qiskit_aer/backends/aer_compiler.py

```diff
@@ -199,12 +215,78 @@ def _inline_circuit(self, circ, continue_label, break_label, bit_map=None):
 
         return ret
 
-    def _convert_c_if_args(self, cond_tuple, bit_map):
+    class _ClbitConverter(ExprVisitor):
+        """Apply cbit index conversion in Expr tree"""
+
+        def __init__(self, bit_map):
+            self.bit_map = bit_map
+
+        def visit_value(self, node, /):
+            # pylint: disable=unused-variable
+            return node
+
+        def visit_var(self, node, /):
+            if isinstance(node.var, Clbit):
+                node.var = self.bit_map[node.var]
+            else:
+                node.var = [self.bit_map[clbit] for clbit in node.var]
+            return node
+
+        def visit_cast(self, node, /):
+            node.operand.accept(self)
+            return node
+
+        def visit_unary(self, node, /):
+            node.operand.accept(self)
+            return node
+
+        def visit_binary(self, node, /):
+            node.left.accept(self)
+            node.right.accept(self)
+            return node
+
+        def visit_generic(self, node, /):
+            raise AerError(f"unsupported expression is used: {node.__class__}")
+
+    def _convert_jump_conditional(self, cond_tuple, bit_map):
         """Convert a condition tuple according to the wire map."""
-        if isinstance(cond_tuple[0], Clbit):
+        if isinstance(cond_tuple, Expr):
+            return cond_tuple
+        elif isinstance(cond_tuple[0], Clbit):
             return (bit_map[cond_tuple[0]], cond_tuple[1])
-        # ClassicalRegister conditions should already be in the outer circuit.
-        return cond_tuple
+        elif isinstance(cond_tuple[0], ClassicalRegister):
+            # ClassicalRegister conditions should already be in the outer circuit.
+            return cond_tuple
+
+        # left and right expression
+        def _convert_clbit_index(expr):
+            if isinstance(expr, Unary):
+                _convert_clbit_index(expr.operand)
+            elif isinstance(expr, Binary):
+                _convert_clbit_index(expr.left)
+                _convert_clbit_index(expr.right)
+            elif isinstance(expr, Var):
+                if isinstance(expr.var, Clbit):
+                    expr.var = bit_map[expr.var]
+                else:
+                    expr.var = [bit_map[clbit] for clbit in expr.var]
+            return expr
+
+        left = deepcopy(cond_tuple[0]).accept(AerCompiler._ClbitConverter(bit_map))
+        right = cond_tuple[1]
+        return (left, right)
```

Comment:
I have a previous review comment about this up top, but in what situations do we reach line 260?  As far as I know, a condition can _only_ be an `Expr` or a `tuple[Clbit | ClassicalRegister, int]`, so the if statements on lines 253 to 239 should be exhaustive, right?  In what situations might we get through them?

Reply:
Thanks. `Var` is possible to come here. I rewrite this part.